extensions/kubernetes-client/runtime/src/main/java/io/quarkus/kubernetes/client/runtime/KubernetesClientBuildConfig.java

```diff
@@ -0,0 +1,148 @@
+
+    /**
+     * Whether or not the client should trust a self signed certificate if so presented by the API server
+     */
+    @ConfigItem(defaultValue = "false")
+    boolean trustCerts;
+
+    /**
+     * URL of the Kubernetes API server
+     */
+    @ConfigItem
+    String masterUrl;
+
+    /**
+     * Default namespace to use
+     */
+    @ConfigItem
+    String namespace;
+
+    /**
+     * CA Certificate file
+     */
+    @ConfigItem
+    String caCertFile;
+
+    /**
+     * CA Certificate data
+     */
+    @ConfigItem
+    String caCertData;
+
+    /**
+     * Client certificate file
+     */
+    @ConfigItem
+    String clientCertFile;
+
+    /**
+     * Client certificate data
+     */
+    @ConfigItem
+    String clientCertData;
+
+    /**
+     * Client key file
+     */
+    @ConfigItem
+    String clientKeyFile;
+
+    /**
+     * Client key data
+     */
+    @ConfigItem
+    String clientKeyData;
+
+    /**
+     * Client key algorithm
+     */
+    @ConfigItem
+    String clientKeyAlgo;
+
+    /**
+     * Client Key passphrase
+     */
+    @ConfigItem
+    String clientKeyPassphrase;
+
+    /**
+     * Kubernetes auth username
+     */
+    @ConfigItem
+    String username;
+
+    /**
+     * Kubernetes auth password
+     */
+    @ConfigItem
+    String password;
+
+    /**
+     * Watch reconnect interval in milliseconds
+     */
+    @ConfigItem(defaultValue = "1000") // default lifted from Kubernetes Client
+    Integer watchReconnectInterval;
+
+    /**
+     * Maximum reconnect attempts in case of watch failure
+     * By default there is no limit to the number of reconnect attempts
+     */
+    @ConfigItem(defaultValue = "-1") // default lifted from Kubernetes Client
+    Integer watchReconnectLimit;
+
+    /**
+     * Maximum amount of time in milliseconds to wait for a connection with the API server to be established
+     */
+    @ConfigItem(defaultValue = "10000") // default lifted from Kubernetes Client
+    Integer connectionTimeout;
+
+    /**
+     * Maximum amount of time in milliseconds to wait for a to the API server to be completed
+     */
+    @ConfigItem(defaultValue = "10000") // default lifted from Kubernetes Client
+    Integer requestTimeout;
+
+    /**
+     * Maximum amount of time in milliseconds to wait for a rollout to be completed
+     */
+    @ConfigItem(defaultValue = "900000") // default lifted from Kubernetes Client
+    Long rollingTimeout;
+
+    /**
+     * http proxy used to access the Kubernetes API server
+     */
+    @ConfigItem
+    String httpProxy;
+
+    /**
+     * https proxy used to access the Kubernetes API server
+     */
+    @ConfigItem
+    String httpsProxy;
+
+    /**
+     * proxy username
+     */
+    @ConfigItem
+    String proxyUsername;
+
+    /**
+     * proxy password
+     */
+    @ConfigItem
+    String proxyPassword;
+
+    /**
+     * ips or hosts to exclude from proxying
+     */
+    @ConfigItem
+    String[] noProxy;
```

Comment:
Maybe let's make it a `List`, maybe? Not too fond of using arrays if we're not performance constrained.

Reply:
Makes sense